coordinator/src/test/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlannerSpec.scala

```diff
@@ -997,4 +997,30 @@ class ShardKeyRegexPlannerSpec extends AnyFunSpec with Matchers with ScalaFuture
       case _ => fail("Expected two children")
     }
   }
-}
\ No newline at end of file
+
+  it("should materialize absent function mapper correctly with implicit WS") {
```

Comment:
nit: Can we add one more test for absent without over_time function .. e.x `absent(test{_ns_ = \"App-1\", instance = \"Inst-1\" })` This should return column filters with _ws_ included.

Reply:
Wait, I don't think that's right. Absent either returns an empty result or a single RangeVector with an empty key.

Reply:
I agree that we should add similar test cases for `absent` and `absent_over_time`, though.

Reply:
I think just asserting that these should give empty column filters is sufficient:
```
absent(test{_ws_=\"demo\", _ns_ = \"App-1\", instance = \"Inst-1\" })
absent(test{_ns_ = \"App-1\", instance = \"Inst-1\" })
absent_over_time(test{_ws_=\"demo\", _ns_ = \"App-1\", instance = \"Inst-1\" }[5m])
absent_over_time(test{_ns_ = \"App-1\", instance = \"Inst-1\" }[5m])
```

(and I would be curious to know if anything different happens...)

Reply:
Correction to the above: if there are non-metric labels given with the query, they are included in the absent result.

The prometheus docs show the following:
- absent(<selector>): all non-metric labels should be in the result iff rows are present
- absent_over_time(<range-selector>): all non-metric labels should be in the result iff rows are present
- absent(sum_over_time(<range-selector>)): `sum_over_time` will either return rows (with full keys), or not. If it does, absent returns nothing. If it does not, then there are no selector labels to even include as a RangeVector key.

I'll approve the PR, since these look good. Could you add some functional tests for each of these cases just to be safe? (I see only one `absent(sum_over_time)` test, for example).